### PR TITLE
Draft: Add RequestErrorContext and error propagation infrastructure for request aborts

### PR DESCRIPTION
**Pull request by wseaton**
This ties in with: #26171 

PoC of an error context and propagation system that allows exposing rich error context and metadata back up through the API server.

### Example implementations

**1. KV Transfer Load Failures**
```python
# When remote KV cache blocks fail to load
error_ctx = RequestErrorContext(
    error_type=ErrorType.KV_TRANSFER_FAILURE,
    message="Failed to load KV cache from remote source",
    http_status=HTTPStatus.BAD_GATEWAY,
    metadata={"affected_tokens": num_tokens, "invalid_blocks": len(invalid_block_ids)} #in addition everything in kv_transfer_params, eg. remote_host_id for routing purposes
)
# Client receives 502 with actionable details instead of generic 500
```
2. Resource Exhaustion (scheduler preemption handling)
```py
# When request aborted after excessive preemptions
error_ctx = RequestErrorContext(
    error_type=ErrorType.RESOURCE_EXHAUSTION,
    message="Request aborted due to memory pressure",
    http_status=HTTPStatus.SERVICE_UNAVAILABLE,
    metadata={"num_preemptions": request.num_preemptions, "available_blocks": blocks}
)
# Client receives 503, can implement retry with backoff
```
3. Corrupted Model Output
```py
# When NaN detected in model logits
if request.is_output_corrupted:
    error_ctx = RequestErrorContext(
        error_type=ErrorType.MODEL_OUTPUT_CORRUPTED,
        message="Model generated corrupted output (NaN in logits)",
        http_status=HTTPStatus.INTERNAL_SERVER_ERROR,
        metadata={"num_nans": request.num_nans_in_logits, "position": request.num_output_tokens}
    )
```

### CODE REVIEW

**Comment by njhill** [comment]
I could be convinced otherwise but I don't feel that this is needed... details of runtime errors that occur within the engine should be logged but I'm not sure should be propagated back to the external client, since it implies an internal error and security-wise such details shouldn't be exposed to users.

These errors require administrator intervention and not. If the error implies the server is in an unstable or broken state then it should shut itself down.

**Comment by wseaton** [comment]
@njhill I agree with you in the general sense where vLLM's API server is more directly exposed, but KV Connector failures in a P/D disaggregated setup are a little unique and might warrant this enhanced handling. For P/D I'm thinking of an IGW-style proxy sitting in front of vLLM that acts as the security boundary. This proxy would:

- strip any sensitive KV details before they reach actual clients
- use the structured errors internally to make smarter routing decisions (like whether to retry elsewhere or fail fast)
- convert to appropriate HTTP codes (502 vs 503) based on the actual failure type

Related:
* how vllm handles aborts today (using`finish_reason=abort`) is [not OpenAI spec compliant](https://github.com/openai/openai-openapi/blob/manual_spec/openapi.yaml#L19675-L19696), and popular clients don't really know how to handle them. Looking at fixing this was actually what drove me to research this

cc @bbrowning
